lib/awful/hotkeys_popup/widget.lua

```diff
@@ -357,21 +437,21 @@ function widget.new()
         local widget_obj = {}
         widget_obj.current_page = 1
         widget_obj.wibox = mywibox
-        function widget_obj:page_next()
-            if self.current_page == #pages then return end
-            self.current_page = self.current_page + 1
-            self.wibox:set_widget(pages[self.current_page])
+        function widget_obj.page_next(_self)
+            if _self.current_page == #pages then return end
+            _self.current_page = _self.current_page + 1
+            _self.wibox:set_widget(pages[_self.current_page])
         end
-        function widget_obj:page_prev()
-            if self.current_page == 1 then return end
-            self.current_page = self.current_page - 1
-            self.wibox:set_widget(pages[self.current_page])
+        function widget_obj.page_prev(_self)
+            if _self.current_page == 1 then return end
+            _self.current_page = _self.current_page - 1
+            _self.wibox:set_widget(pages[_self.current_page])
         end
-        function widget_obj:show()
-            self.wibox.visible = true
+        function widget_obj.show(_self)
```

Comment:
Why the change? If it was about shadowing `self`, then you should consider moving these methods out of the constructor.